sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamSuite.scala

```diff
@@ -81,4 +85,62 @@ class StreamSuite extends StreamTest with SharedSQLContext {
       AddData(inputData, 1, 2, 3, 4),
       CheckAnswer(2, 4))
   }
+
+  test("DataFrame reuse") {
+    def assertDF(df: DataFrame) {
+      withTempDir { outputDir =>
+        withTempDir { checkpointDir =>
+          val query = df.write.format("parquet")
+            .option("checkpointLocation", checkpointDir.getAbsolutePath)
+            .startStream(outputDir.getAbsolutePath)
+          try {
+            eventually(timeout(streamingTimeout)) {
+              val outputDf = sqlContext.read.parquet(outputDir.getAbsolutePath).as[Long]
+              checkDataset[Long](outputDf, (0L to 10L).toArray: _*)
+            }
+          } finally {
+            query.stop()
+          }
+        }
+      }
+    }
+
+    val df = sqlContext.read.format(classOf[FakeDefaultSource].getName).stream()
+    assertDF(df)
+    assertDF(df)
+    assertDF(df)
```

Comment:
I was saying that when I ran it locally it passed (without the other fixes in this PR) when there were only two.